server/scheduling/scheduler.ts

```diff
@@ -0,0 +1,39 @@
+import { initManagers } from '../db/initDB';
+import { IPost } from '../db/models';
+import mongoose = require('mongoose');
+
+// eslint-disable-next-line no-unused-vars
+const { jobManager, postManager } = initManagers();
+const lt = require('long-timeout');
+
+// run all the tasks which are in the database
+(async() => {
+	(await jobManager.getAllJobs()).forEach((elem) => {
+		if (dateDiff(new Date(), elem.data.date) < 0) {
+			jobManager.deleteJob(elem._id);
+		} else {
+			// eslint-disable-next-line no-new-func
+			const fn = new Function('return ' + elem.value)();
```

Comment:
What this function means and how it should work?

Reply:
This is similar to eval where I just convert the string function elem.value to an actual function I can call. The reason I use the new Function construct is that it suffers fewer performance issues than eval.